cmd/kubeadm/app/cmd/join.go

```diff
@@ -204,20 +206,31 @@ func (j *Join) Run(out io.Writer) error {
 		return err
 	}
 
-	hostname := nodeutil.GetHostname(j.cfg.NodeName)
-
 	client, err := kubeconfigutil.KubeConfigToClientSet(cfg)
 	if err != nil {
 		return err
 	}
 	if err := kubeadmnode.ValidateAPIServer(client); err != nil {
 		return err
 	}
-	if err := kubeadmnode.PerformTLSBootstrap(cfg, hostname); err != nil {
-		return err
+
+	kubeconfigFile := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.KubeletBootstrapKubeConfigFileName)
+
+	// Depending on the kubelet version, we might perform the TLS bootstrap or not
+	kubeletVersionBytes, err := exec.Command("sh", "-c", "kubelet --version").Output()
+	// In case the command executed successfully and returned v1.7-something, we'll perform TLS Bootstrapping
+	// Otherwise, just assume v1.8
+	// TODO: In the beginning of the v1.9 cycle, we can remove the logic as we then don't support v1.7 anymore
```

Comment:
Side comment but it might be nice to keep a version check in here permanently to validate the kubeadm<>kubelet skew is supported and fail fast with a nice error if not.

Reply:
I'm planning to add such a version check in v1.9
Too late for v1.8 indeed

Reply:
Filed https://github.com/kubernetes/kubeadm/issues/430 so we don't forget.

Reply:
Thanks!